pkg/backend/filestate/state.go

```diff
@@ -299,19 +298,19 @@ func (b *localBackend) getHistory(name tokens.QName) ([]backend.UpdateInfo, erro
 
 	var updates []backend.UpdateInfo
 
-	// os.ReadDir returns the array sorted by file name, but because of how we name files, older updates come before
+	// listBucket returns the array sorted by file name, but because of how we name files, older updates come before
 	// newer ones. Loop backwards so we added the newest updates to the array we will return first.
 	for i := len(allFiles) - 1; i >= 0; i-- {
 		file := allFiles[i]
-		filepath := path.Join(dir, file.Name())
+		filepath := path.Join(dir, objectName(file))
```

Comment:
Is this needed? IIUC, `path.Join(dir, objectName(file))` == `file.Key`...?

Reply:
Looks to be that way.  Thanks!